pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,223 @@ func TestAccessTokenFile(t *testing.T) {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite2"
-			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
-
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
-
-			// Second time
-			dataActual = []byte{'t', 'e', 's', 't', '2'}
-			err = tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
 
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
-
-			// Test we are reading the new Data
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
-
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+			Convey("When running the Write method", func() {
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Writes the access token without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+			})
+
+			Convey("When running the Write method a second time", func() {
+				dataActual = []byte{'t', 'e', 's', 't', '2'}
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Writes the data the file as expected", func() {
+					dataExpected, _ := tokenInFile.Read()
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 	})
 
 	Convey("Delete", t, func() {
-		Convey("Returns no error after read write and delete", func() {
-			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
-			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
-
-			dataFromRead, err := tokenInFile.Read()
-			So(err, ShouldEqual, nil)
-
-			err = tokenInFile.Delete()
-			So(err, ShouldEqual, nil)
 
-			// Check that file does not exits
-			_, err = os.Stat("/tmp/accessTokenFileDel1")
-			So(err, ShouldNotEqual, nil)
+		Convey("Given access token with data", func() {
+			dataActual := []byte{'t', 'e', 's', 't'}
 
-			// Both input & output arrays should be cleared from memory
-			empty := make([]byte, len(dataActual))
-			eq := reflect.DeepEqual(dataActual, empty)
-			So(eq, ShouldEqual, true)
-			eq = reflect.DeepEqual(dataFromRead, empty)
-			So(eq, ShouldEqual, true)
+			Convey("When running the Write method", func() {
+				tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("When running the Read method", func() {
```

Comment:
why do we need this? we're not testing here the `read` method. if we do need it, it should be Given (but i don't think we need it anyway)

Reply:
why don't need this in a `convey` or just not all? Nessi wanted to test end-to-end with write, read, delete so that's why we wanted the read method in there but we can probably not worry about the testing if there is an err because that should be taken care of when `Read` is actually tested above

Reply:
yes, i would remove `read` from this test